readthedocs/core/utils/__init__.py

```diff
@@ -112,7 +111,21 @@ def prepare_build(
         'commit': commit,
     }
 
-    if record:
+    if build:
+        build.state = BUILD_STATE_TRIGGERED
+        build.success = True
+        build.commit = commit
+        build.commands.all().delete()
+        build.save()
+        kwargs['build_pk'] = build.pk
+
+        # Show a step in the UI that the build is being rebuilt
+        BuildCommandResult.objects.create(
+            build=build, command='Rebuilding build', exit_code=0,
+            start_time=datetime.utcnow(), end_time=datetime.utcnow(),
+        )
```

Comment:
I'd probably remove this entirely, I don't think it's worth introducing UI confusion in the command output. The best place to solve this would be either popping up a confirmation box for the user on click, or simply revert the state back to show the UI as "loading". Either would communicate to the user that the build is restarting

Reply:
I'm more worried about other folks looking at it and being confused about why it was running again after it had already built. I think we don't do a great job of showing history on builds -- we hit this issue with failures & retries as well, so we need some kind of pattern for it.

Reply:
I've updated this to use the new build status, which is a better solution 👍